midend/parserUnroll.cpp

```diff
@@ -367,6 +366,41 @@ class ParserSymbolicInterpreter {
             auto mc = sord->to<IR::MethodCallStatement>();
             auto e = ev.evaluate(mc->methodCall, false);
             success = reportIfError(state, e);
+        } else if (auto bs = sord->to<IR::BlockStatement>()) {
+            IR::IndexedVector<IR::StatOrDecl> newComponents;
+            for (auto* component : bs->components) {
+                auto newComponent = executeStatement(state, component, valueMap);
+                if (!newComponent)
+                    success = false;
+                else
+                    newComponents.push_back(newComponent);    
+            }
+            sord = new IR::BlockStatement(newComponents);
+        } else if (sord->is<IR::IfStatement>()) {
+            auto ifs = sord->to<IR::IfStatement>()->clone();
+            auto ifcond = ev.evaluate(ifs->condition, true);
+            success = reportIfError(state, ifcond);
+            if (success) {
+                const IR::Statement* ifComponent = nullptr;
+                if (ifcond != nullptr) {
```

Comment:
We likely do not need this check as previously discussed. There is no way ifcond can be a nullptr since this triggers a bug. 
`auto ifcond = ev.evaluate(ifs->condition, true);`
should be 
`const auto *ifcond = ev.evaluate(ifs->condition, true);`
to make clear that ifcond can not be modified by methods such as reportIfError.